sphinx/ext/autodoc/directive.py

```diff
@@ -82,12 +83,7 @@ def process_documenter_options(documenter, config, options):
 
 def parse_generated_content(state, content, documenter):
     # type: (State, StringList, Documenter) -> List[nodes.Node]
-    try:
-        # use a custom reporter that correctly assigns lines to source
-        # filename/description and lineno
-        old_reporter = state.memo.reporter
-        state.memo.reporter = AutodocReporter(content, state.memo.reporter)
-
+    with switch_source_input(state, content):
```

Comment:
Could you re-add the comments you had here? For someone not familiar with the code, it's pretty tough to figure out what's going on without said comments